File: src/skyjam/common/grid.py
"""Sampling grid: which points to poll to cover the area of interest.

The API is queried per point with a radius, so the region of interest is covered
by a set of overlapping discs. Points are chosen to cover European airspace plus
the regions where GNSS interference is routinely reported (Baltic, Black Sea,
Eastern Mediterranean), with control regions in Western Europe that let us show
the model is not simply memorising "east is bad".
"""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class SamplePoint:
    name: str
    lat: float
    lon: float
    radius_nm: int = 250
    #: Regions we expect to be quiet; kept explicitly so the class balance and
    #: the false-positive rate can be reported per region.
    control: bool = False


# Roughly 250 NM ~ 463 km radius per disc. Spacing is deliberately tighter than
# the diameter so neighbouring discs overlap and no corridor falls between them.
SAMPLE_POINTS: tuple[SamplePoint, ...] = (
    # --- Reported interference regions ---
    SamplePoint("baltic_kaliningrad", 54.7, 20.5),
    SamplePoint("gulf_of_finland", 59.4, 24.8),
    SamplePoint("baltic_south", 55.5, 16.5),
    SamplePoint("finland_north", 65.5, 26.0),
    SamplePoint("black_sea_west", 44.5, 31.0),
    SamplePoint("black_sea_east", 43.0, 38.0),
    SamplePoint("eastern_med", 34.8, 33.5),
    SamplePoint("levant_north", 36.5, 36.5),
    SamplePoint("caucasus", 41.0, 44.0),
    SamplePoint("poland_east", 52.2, 23.0),
    SamplePoint("romania_moldova", 46.5, 27.5),
    # --- Control regions, expected quiet ---
    SamplePoint("alps_switzerland", 47.0, 8.3, control=True),
    SamplePoint("uk_south", 51.5, -0.5, control=True),
    SamplePoint("iberia_central", 40.4, -3.7, control=True),
    SamplePoint("france_central", 46.5, 2.5, control=True),
    SamplePoint("germany_central", 50.5, 9.5, control=True),
    SamplePoint("italy_central", 42.5, 12.5, control=True),
    SamplePoint("scandinavia_west", 59.9, 10.7, control=True),
)


def active_points(include_controls: bool = True) -> tuple[SamplePoint, ...]:
    if include_controls:
        return SAMPLE_POINTS
    return tuple(p for p in SAMPLE_POINTS if not p.control)

File: services/ingestor/internal/grid/grid.go
// Package grid defines which points are polled to cover the area of interest.
//
// This mirrors src/skyjam/common/grid.py exactly. The duplication is
// deliberate and bounded: it is a list of coordinates, not a modelling rule.
// Label semantics are never duplicated into Go, because that is where
// training/serving skew actually comes from. A parity test keeps the two
// lists honest.
package grid

// SamplePoint is one polling disc.
type SamplePoint struct {
	Name     string
	Lat      float64
	Lon      float64
	RadiusNM int
	// Control marks regions expected to be quiet, so class balance and
	// false-positive rate can be reported per region.
	Control bool
}

// SamplePoints covers European airspace plus the regions where GNSS
// interference is routinely reported, with Western-European controls.
// Spacing is tighter than the disc diameter so no corridor falls between them.
var SamplePoints = []SamplePoint{
	// --- Reported interference regions ---
	{Name: "baltic_kaliningrad", Lat: 54.7, Lon: 20.5, RadiusNM: 250},
	{Name: "gulf_of_finland", Lat: 59.4, Lon: 24.8, RadiusNM: 250},
	{Name: "baltic_south", Lat: 55.5, Lon: 16.5, RadiusNM: 250},
	{Name: "finland_north", Lat: 65.5, Lon: 26.0, RadiusNM: 250},
	{Name: "black_sea_west", Lat: 44.5, Lon: 31.0, RadiusNM: 250},
	{Name: "black_sea_east", Lat: 43.0, Lon: 38.0, RadiusNM: 250},
	{Name: "eastern_med", Lat: 34.8, Lon: 33.5, RadiusNM: 250},
	{Name: "levant_north", Lat: 36.5, Lon: 36.5, RadiusNM: 250},
	{Name: "caucasus", Lat: 41.0, Lon: 44.0, RadiusNM: 250},
	{Name: "poland_east", Lat: 52.2, Lon: 23.0, RadiusNM: 250},
	{Name: "romania_moldova", Lat: 46.5, Lon: 27.5, RadiusNM: 250},
	// --- Control regions, expected quiet ---
	{Name: "alps_switzerland", Lat: 47.0, Lon: 8.3, RadiusNM: 250, Control: true},
	{Name: "uk_south", Lat: 51.5, Lon: -0.5, RadiusNM: 250, Control: true},
	{Name: "iberia_central", Lat: 40.4, Lon: -3.7, RadiusNM: 250, Control: true},
	{Name: "france_central", Lat: 46.5, Lon: 2.5, RadiusNM: 250, Control: true},
	{Name: "germany_central", Lat: 50.5, Lon: 9.5, RadiusNM: 250, Control: true},
	{Name: "italy_central", Lat: 42.5, Lon: 12.5, RadiusNM: 250, Control: true},
	{Name: "scandinavia_west", Lat: 59.9, Lon: 10.7, RadiusNM: 250, Control: true},
}
